### PR TITLE
adding set_rts and trying to execute modbus_rtu_set_custom_rts

### PR DESCRIPTION
**Pull request by fel115**

### CODE REVIEW

**Comment by fel115** [comment]
For the moment I have problems with using a function as function argument.

**Comment by fel115** [comment]
@markus2330, do you have any suggestions on how I could pass a function as an argument? My IDE indicates that I need to use `Option<fn(*mut _modbus, i32)>` as the parameter type.

**Comment by markus2330** [comment]
Unfortunately I also didn't need callback functions in Rust yet. Maybe @kitzbergerg can give us a quick hint?

**Comment by kitzbergerg** [comment]
I have never done it myself, but there are resources.
Maybe something like the following is what you are looking for?
- https://doc.rust-lang.org/nomicon/ffi.html#callbacks-from-c-code-to-rust-functions (This whole section has quite good explanations about FFI)

If you are generally doing more low level stuff and might need a bit of unsafe I would suggest reading the whole Rustonomicon.

**Comment by fel115** [comment]
@kitzbergerg thank you for your response! I genuinely appreciate your willingness to help. Although I haven't tried it myself, the resources you provided seem to be exactly what I was looking for. The link you shared, https://doc.rust-lang.org/nomicon/ffi.html#callbacks-from-c-code-to-rust-functions, looks promising, and I'm excited to explore it further. Your suggestion to read the whole Rustonomicon for more low-level stuff with `unsafe` sounds like a great idea too. Thanks again for your assistance and valuable insights. If I have any more questions, I'll definitely reach out.

**Comment by markus2330** [comment]
Great :revolving_hearts: 

> Your suggestion to read the whole Rustonomicon for more low-level stuff with unsafe sounds like a great idea too.

I hope all the unsafe code, if needed at all, can be hidden in libmodbus-rs, would be great to not have any unsafe code in Opensesame.

**Comment by fel115** [comment]
It appears to be working; I tested it with the weather station, and I received the same result as with the C-program.

**Review by markus2330 — APPROVED** [review]
LGTM, I don't think we can push this upstream but for us this should be now fine.

Main "problem": Why do you use gpio? Afaik only gpio-cdev uses the latest Linux API (the old one being deprecated/removed soon) and has async-tokio features. But as this is hidden here in libmodbus-rs, it hopefully should be fine for now.

**Comment by fel115** [comment]
> Main "problem": Why do you use gpio? Afaik only gpio-cdev uses the latest Linux API (the old one being deprecated/removed soon) and has async-tokio features. But as this is hidden here in libmodbus-rs, it hopefully should be fine for now.

Yeah, I can switch it to gpio-cdev, I don't think that should be a big problem.

File: src/modbus_rtu.rs [hunk 1]
```diff
@@ -305,7 +308,35 @@ impl ModbusRTU for Modbus {
     ///
     /// TODO: implement rtu_set_custom_rts()!
     fn rtu_set_custom_rts(&mut self, _mode: RequestToSendMode) -> Result<i32, Error> {
-        unimplemented!()
+        //I need to distinguish between RtuRtsUp and RtuRtsDown, and now it's for RtuRtsUp."
+        unsafe extern "C" fn custom_rts_callback(_ctx: *mut ffi::modbus_t, on: c_int) {
+            let mut chip = Chip::new("/dev/gpiochip0").expect("Failed to open gpiochip0");
```
Comment (markus2330) [inline src/modbus_rtu.rs:313]: Not ideal that this gets reopened every time, can you pass this somehow via an handle?

**Review by markus2330 — APPROVED** [review]
LGTM

**Pull request merged**